### DISCUSSION TITLE
conditional preprocessor syntax highlighting in C language

### DISCUSSION THREAD

**Discussion by masriomarm** [helix-editor/helix#8406] [Q&A]
How to highlight inactive regions as greyed out like the image attached? I'm using clang.
![inactive_region](https://github.com/helix-editor/helix/assets/77388732/529c3c96-d54c-46a6-9d4f-00d73091a2b9)

**Comment by the-mikedavis** [comment]
Syntax highlighting is done with tree-sitter which doesn't have semantic information about these blocks, so turning off highlighting depending on your machine/architecture for example isn't possible

**Reply by masriomarm** [reply]
I've read your reply few times until I realized the image content. Apologies for the confusion.
I just need to grey out sections of code that are inactive for example
```c
#undef foo
#define bar
#ifdef foo
// this area syntax highlighting should be greyed out
#endif // foo

#ifdef bar
// this are syntax highlighting should be normal
#endif // bar
```
`clangd` is aware of the inactive sections, since it refuse to jump to deceleration of symbols that are conditionally defined. VSCode has a similar feature, I think it's a `clangd` client side dependent feature.

**Reply by A-Walrus** [reply]
In helix syntax highlighting is handled entirely by tree-sitter, so the LSP (in this case clang) has no effect.
In vscode the plugin takes into account information from clangd to determine highlighting
